File: pinyougou-manager-interface/src/main/java/com/pinyougou/manager/service/BrandService.java
package com.pinyougou.manager.service;

import java.util.List;

import com.pinyougou.common.PageResult;
import com.pinyougou.pojo.TbBrand;

public interface BrandService {

	/**
	 * 查询品牌列表
	 * @return
	 */
	List<TbBrand> findAll();

	/**
	 * 好习惯：在接口上加注释：
	 * 分页查询品牌列表
	 * @param page
	 * @param size
	 * @return
	 */
	public PageResult findPage(Integer page, Integer size);

	/**
	 * 添加品牌
	 * @param brand
	 */
	void add(TbBrand brand);
	
	/**
	 * 根据品牌ID查询品牌对象
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);

	/**
	 * 修改品牌
	 * @param brand
	 */
	void update(TbBrand brand);

	/**
	 * 根据多个ID删除品牌
	 * @param ids
	 */
	void dele(Long[] ids);

	/**
	 * 根据条件对象分页
	 * @param page
	 * @param size
	 * @param brand
	 * @return
	 */
	PageResult search(Integer page, Integer size, TbBrand brand);
	
}

File: pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/service/brandService.js
app.service('brandService',function($http){
	
	this.findAll=function(){
		return $http.get('../brand/findAll');
	}
	this.findPage=function(page,size){
		return $http.get('../brand/findPage?page='+page+'&size='+size);
	}
	this.update=function(entity){
		return $http.post('../brand/update',entity);
	}
	this.add=function(entity){
		return $http.post('../brand/add',entity);
	}
	this.findOne=function(id){
		return $http.get('../brand/findOne?id='+id);
	}
	this.dele=function(ids){
		return $http.get('../brand/dele?ids='+ids);
	}
	this.search=function(page,size,searchEntity){
		//调用后台的接口URL
		return $http.post("../brand/search?page="+page+"&size="+size,searchEntity);
	}
	this.findBrandList=function(){
		return $http.get("../brand/findBrandList");
	}
});